Spring-Boot/Spring-Boot-Student/src/main/java/tw/waterball/judgegirl/springboot/student/controllers/StudentController.java

```diff
@@ -1,98 +1,143 @@
+@RestController
+@AllArgsConstructor
 public class StudentController {
-    private LegacyStudentAPI studentAPI;
-    private TokenService tokenService;
+    private final SignInUseCase signInUseCase;
+    private final SignUpUseCase signUpUseCase;
+    private final GetStudentUseCase getStudentUseCase;
+    private final AuthUseCase authUseCase;
+    private final TokenService tokenService;
 
-    @Autowired
-    public StudentController(LegacyStudentAPI studentAPI, TokenService tokenService) {
-        this.studentAPI = studentAPI;
-        this.tokenService = tokenService;
+    @PostMapping("/signUp")
+    public StudentView signUp(@RequestBody SignUpUseCase.Request request) {
+        SignUpPresenter presenter = new SignUpPresenter();
+        signUpUseCase.execute(request, presenter);
+        return presenter.present();
     }
 
-    @PostMapping(path = "/auth")
-    public ResponseEntity auth(@RequestHeader("Authorization") String authorization) {
+    @PostMapping("/login")
+    public LoginResponse login(@RequestBody SignInUseCase.Request request) {
+        SignInPresenter presenter = new SignInPresenter(tokenService);
+        signInUseCase.execute(request, presenter);
+        return presenter.present();
+    }
+
+    @GetMapping("{studentId}")
+    public StudentView getStudentById(@PathVariable Integer studentId, @RequestHeader("Authorization") String authorization) {
         String tokenString = HttpHeaderUtils.parseBearerToken(authorization);
-        try {
-            TokenService.Token token = tokenService.parseAndValidate(tokenString);
-            Optional<Student> studentOptional = studentAPI.getStudentById(token.getStudentId());
-            if (studentOptional.isPresent()) {
-                return ResponseEntity.ok(new LoginResponse(token.getStudentId(), studentOptional.get().getAccount(),
-                        token.getToken(), token.getExpiration().getTime()));
-            }
-            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
-        } catch (TokenInvalidException err) {
-            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
-        }
-    }
-
-    @PostMapping(path = "/login")
-    public LoginResponse login(@RequestParam String account, @RequestParam String password) {
-        int studentId = studentAPI.authenticate(account, password);
-        TokenService.Token token = tokenService.createToken(new TokenService.Identity(studentId));
-        return new LoginResponse(studentId, account, token.toString(), token.getExpiration().getTime());
-    }
-
-    @GetMapping("/{studentId}")
-    public ResponseEntity<Student> getStudentById(@PathVariable int studentId,
-                                                  @RequestHeader("Authorization") String authorization) {
+        GetStudentByIdPresenter presenter = new GetStudentByIdPresenter();
+        getStudentUseCase.execute(new GetStudentUseCase.Request(studentId, tokenString), presenter);
```

Comment:
**Authentication** should be performed in the controller, don't delegate it to `GetStudentUseCase`.